internal/client/api_account_test.go

```diff
@@ -2,293 +2,197 @@ package client_test
+	if resp, err := c.ApiReadAccount(ctx, acc); err != nil {
+		t.Errorf("unexpected error reading account: %s", err.Error())
+	} else if !reflect.DeepEqual(expectedAcc, resp) {
+		t.Errorf("expected resp: (%+v),\n got (%+v)", expectedAcc, resp)
 	}
 }
 
 func TestUpdateAccountSuccess(t *testing.T) {
-	uAcc := client.ResponseAccount{
+	ctx := context.Background()
+	auth := commonTestAuth
+
+	acc := "testuser"
+	expectedAcc := client.ResponseAccount{
 		Name: "fakeacc",
 	}
-	mUAcc, err := json.Marshal(uAcc)
-	if err != nil {
-		t.Fatal(err)
-	}
-	tc := testAccountStruct{
-		server: httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
-			w.WriteHeader(http.StatusOK)
-			if _, err := w.Write(mUAcc); err != nil {
-				t.Error(err)
-				return
-			}
-		})),
-		expectedResponse: uAcc,
-		expectedErr:      nil,
-	}
-	defer tc.server.Close()
-	testClient, err := client.NewUnsafeSSLClient(tc.server.URL, "fakeuser", "fakepass")
+
+	svr := NewMockTestServer(&auth, t)
+	svr.AddHandler(http.MethodPatch, fmt.Sprintf("%s/%s", client.URLTargetForAccounts, acc), MockServerHandlerGeneratorReturnJson(expectedAcc))
+	defer svr.Close()
+
+	c, err := svr.Client()
 	if err != nil {
-		t.Error("couldn't create test client")
+		t.Fatalf("Could not make a client: %s", err)
 	}
-	ctx := context.Background()
-	resp, err := testClient.ApiUpdateAccount(ctx, "fakeid", client.UpdateAccount{FullName: "mock"})
 
-	if !reflect.DeepEqual(tc.expectedResponse, resp) {
-		t.Errorf("expected resp: (%+v),\n got (%+v)", tc.expectedResponse, resp)
-	}
-	if !errors.Is(err, tc.expectedErr) {
-		t.Errorf("expected error: (%v),\n got (%v)", tc.expectedErr, err)
+	if resp, err := c.ApiUpdateAccount(ctx, acc, client.UpdateAccount{FullName: "mock"}); err != nil {
+		t.Errorf("unexpected error updating account: %s", err.Error())
+	} else if !reflect.DeepEqual(expectedAcc, resp) {
+		t.Errorf("expected resp: (%+v),\n got (%+v)", expectedAcc, resp)
 	}
 }
 
 func TestReadAccountsSuccess(t *testing.T) {
-	resAccs := []client.ResponseAccount{}
-	resAccs = append(resAccs,
-		client.ResponseAccount{Name: "mock1"}, client.ResponseAccount{Name: "mock2"})
-
-	accs := struct {
-		UsersCount    int    `json:"usersCount"`
-		OrgsCount     int    `json:"orgsCount"`
-		ResourceCount int    `json:"resourceCount"`
-		NextPageStart string `json:"nextPageStart"`
-
-		Accounts []client.ResponseAccount `json:"accounts"`
-	}{Accounts: resAccs}
-	mResAccs, err := json.Marshal(accs)
-	if err != nil {
-		t.Fatal(err)
+	ctx := context.Background()
+	auth := commonTestAuth
+
+	readFilter := client.AccountFilterUsers
+	expectedAccs := []client.ResponseAccount{
+		{Name: "mock1"},
+		{Name: "mock2"},
 	}
-	tc := testAccountStruct{
-		server: httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
-			w.WriteHeader(http.StatusOK)
-			if _, err := w.Write(mResAccs); err != nil {
-				t.Error(err)
-				return
-			}
-		})),
-		expectedErr: client.ErrUnmarshaling,
```

Comment:
I am not sure how this even worked. The test is a success-test, but it expects an error? Does that mean that the expected test infra didn't work, or that the api call returned an error on success?

Reply:
I see it.  The if condition throws an error if the err == the expected error for this test.